src/arkode/arkode_lsrkstep.c

```diff
@@ -1593,98 +2205,98 @@ sunbooleantype lsrkStep_CheckNVector(N_Vector tmpl)
 {
-  int retval = SUN_SUCCESS;
-
-  if ((step_mem->is_ext_dom_eig))
+  if (dsm <= ONE)
   {
-    retval = step_mem->extDomEig(ark_mem->tn, ark_mem->ycur, ark_mem->fn,
-                                 &step_mem->lambdaR, &step_mem->lambdaI,
-                                 ark_mem->user_data, ark_mem->tempv1,
-                                 ark_mem->tempv2, ark_mem->tempv3);
-    step_mem->num_dom_eig_updates++;
-    if (retval != ARK_SUCCESS)
-    {
-      arkProcessError(ark_mem, ARK_DOMEIG_FAIL, __LINE__, __func__, __FILE__,
-                      "Unable to estimate the dominant eigenvalue");
-      return (ARK_DOMEIG_FAIL);
-    }
+    N_VScale(ONE, ark_mem->tempv2, ark_mem->fn);
+    ark_mem->fn_is_current = SUNTRUE;
 
-    if (step_mem->lambdaR * ark_mem->h > ZERO)
-    {
-      arkProcessError(NULL, ARK_DOMEIG_FAIL, __LINE__, __func__, __FILE__,
-                      "lambdaR*h must be nonpositive");
-      return (ARK_DOMEIG_FAIL);
-    }
-    else if (step_mem->lambdaR == 0 && SUNRabs(step_mem->lambdaI) > 0)
+    step_mem->dom_eig_is_current = (step_mem->const_Jac == SUNTRUE);
+    step_mem->dom_eig_nst = (step_mem->dom_eig_nst + 1) % step_mem->dom_eig_freq;
+    step_mem->dom_eig_update = SUNFALSE;
+    if (step_mem->dom_eig_nst == 0)
     {
-      arkProcessError(NULL, ARK_DOMEIG_FAIL, __LINE__, __func__, __FILE__,
-                      "DomEig cannot be purely imaginary");
-      return (ARK_DOMEIG_FAIL);
+      step_mem->dom_eig_update = !step_mem->dom_eig_is_current;
     }
-
-    step_mem->lambdaR *= step_mem->dom_eig_sfty;
-    step_mem->lambdaI *= step_mem->dom_eig_sfty;
-    step_mem->sprad =
-      SUNRsqrt(SUNSQR(step_mem->lambdaR) + SUNSQR(step_mem->lambdaI));
   }
   else
   {
-    arkProcessError(NULL, ARK_DOMEIG_FAIL, __LINE__, __func__, __FILE__,
-                    "Internal dom_eig is not supported yet!");
-    return (ARK_DOMEIG_FAIL);
+    step_mem->dom_eig_update = !step_mem->dom_eig_is_current;
+    step_mem->dom_eig_nst    = 0;
   }
-  step_mem->dom_eig_is_current = SUNTRUE;
+}
+
+/*---------------------------------------------------------------
+  lsrkStep_ComputeNewDomEig:
+
+  This routine computes new dom_eig and returns SUN_SUCCESS.
+  ---------------------------------------------------------------*/
 
-  step_mem->spr_max = (step_mem->sprad > step_mem->spr_max) ? step_mem->sprad
-                                                            : step_mem->spr_max;
+int lsrkStep_ComputeNewDomEig(ARKodeMem ark_mem, ARKodeLSRKStepMem step_mem)
+{
+  int retval = SUN_SUCCESS;
 
-  if (step_mem->sprad < step_mem->spr_min || ark_mem->nst == 0)
+  retval = step_mem->dom_eig_fn(ark_mem->tn, ark_mem->ycur, ark_mem->fn,
+                                &step_mem->lambdaR, &step_mem->lambdaI,
+                                ark_mem->user_data, ark_mem->tempv1,
+                                ark_mem->tempv2, ark_mem->tempv3);
+  step_mem->dom_eig_num_evals++;
+  if (retval != ARK_SUCCESS)
```

Comment:
Should we allow for recoverable error returns?

Reply:
If fails, we return with the following message "Unable to estimate the dominant eigenvalue", so the user should know what is going on and recover the error if possible. I don't think we explicitly have to return a negative or a positive number, but if you have a strong preference otherwise, I can implement that as well. @gardner48

Reply:
By "recover" I mean cut the time step and reattempt the step. However, I don't see how this would actually help so having any non-zero return be a fatal error seems appropriate.